modules/sov-ibc/src/rpc/context.rs

```diff
@@ -23,8 +24,140 @@ where
     S: Spec,
 {
     /// TODO: Should figure out how can access the proof from the context
-    fn get_proof(&self, _height: Height, _path: &Path) -> Option<Vec<u8>> {
-        Some(vec![])
+    fn get_proof(&self, height: Height, path: &Path) -> Option<Vec<u8>> {
+        let result = match path {
+            Path::ClientState(client_state_path) => {
+                let mut archival_working_set = self
+                    .working_set
+                    .borrow()
+                    .get_archival_at(height.revision_height());
+
+                self.ibc
+                    .client_state_map
+                    .get_with_proof(&client_state_path.0, &mut archival_working_set)
```

Comment:
I noticed that `client_state_map` maps over `ClientId`, rather than `ClientStatePath`. It is correct - but not consistent.

Reply:
Indeed, this originates from ibc-rs, where the `client_state` method traditionally accepts `ClientId`.